dask_expr/expr.py

```diff
@@ -283,6 +283,14 @@ def optimize(self, **kwargs):
     def index(self):
         return Index(self)
 
+    def reset_index(self, drop=False):
+        if is_dataframe_like(self._meta) or is_series_like(self._meta):
+            return ResetIndex(self, drop)
```

Comment:
We can either add `Expr.reset_index` with a `_meta` check like this, **or** we can add `reset_index` to both the `DataFrame` and `Series` APIs. Wasn't sure which was "better" in this case.

Reply:
I'll suggest that we start deferring API to the collections whenever there's any ambiguity.  Are you ok with that convention?

Reply:
Yeah, that works for me.